server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/git/GitRepository.java

```diff
@@ -844,6 +848,24 @@ private static void putChange(Map<String, Change<?>> changeMap, String path, Cha
         assert oldChange == null;
     }
 
+    private static EmptyCommitPolicy getEmptyCommitPolicy(Iterable<Change<?>> changes) {
+        // allMatch returns true if the stream is empty.
+        if (Iterables.isEmpty(changes)) {
+            // Disallowed by default
+            return EmptyCommitPolicy.DISALLOW;
+        }
+        final boolean allJsonPatches =
+                Streams.stream(changes)
+                       .allMatch(change -> change.type() == ChangeType.APPLY_JSON_PATCH);
+        if (allJsonPatches) {
+            // JsonPatch operations may have an empty change that should be ignored.
+            // For example, the removeIfExists operation does nothing if the target path is absent.
```

Comment:
Probably we don't have to do this in this PR but how about distinguishing the ignorable json patch from others?

Reply:
The JSON patch raises an exception if an operation results in no change except for testing operations. So we may not need to distinguish it.

Reply:
Got it. Thanks for the explanation. 👍